packages/cosmic-swingset/test/test-make.js

```diff
@@ -45,18 +55,21 @@ test.serial('make and exec', async t => {
 
 test.serial('integration test: rosetta CI', async t => {
   // Resume the chain... and concurrently, start a faucet AND run the rosetta-cli tests
-  const { pspawnAgd, scenario2 } = t.context;
-
-  t.log('exec agd');
-  t.is(await pspawnAgd([]).exit, 0, 'exec agd exits successfully');
+  const { pspawnMake } = t.context;
 
-  const enoughBlocksToProvision = 15;
+  // Run the chain until error or rosetta-cli exits.
+  const runChain = pspawnMake(['scenario2-run-chain']);
+  const rosetta = pspawnMake(['scenario2-run-rosetta-ci']);
+  const cleanup = async () => {
+    runChain.kill();
+    rosetta.kill();
+    await Promise.allSettled([runChain.exit, rosetta.exit]);
+  };
+  t.teardown(cleanup);
 
-  const queryGrace = 6; // time to query state before shutting down
-  const [_run] = await Promise.all([
-    scenario2.runToHalt({
-      BLOCKS_TO_RUN: enoughBlocksToProvision + queryGrace,
-    }),
-    scenario2.runRosettaCI({}),
+  const code = await Promise.race([
+    rosetta.exit,
+    runChain.exit.then(c => `chain exited unexpectedly with code ${c}`),
```

Comment:
shouldn't this fail the test? or is it too unreliable?

Reply:
I added a comment explaining that we don't want an unhandled rejection if the chain loses the race (normal shutdown behaviour).